### PR TITLE
implement GetTransactions

### PR DESCRIPTION
**Pull request by AmineAfia**
### TL;DR

Refactored transaction and log querying logic to improve code reusability and add support for aggregations.

### What changed?

- Introduced a new `executeQuery` function in the `ClickHouseConnector` to handle both transaction and log queries.
- Added `buildQuery` and `executeAggregateQuery` functions to construct and execute queries more efficiently.
- Updated `GetTransactions` and `GetLogs` methods to use the new query execution logic.
- Modified the `QueryResult` struct to use `[]interface{}` for the `Data` field, allowing for more flexible result types.
- Implemented `scanTransaction` and `scanLog` functions to handle row scanning for each data type.

### How to test?

1. Run existing tests for transaction and log queries to ensure functionality is maintained.
2. Create new test cases that include aggregations in the query filter.
3. Verify that the `QueryResult` structure correctly contains both data and aggregates.
4. Test edge cases with various combinations of filters, sorting, and pagination.

### Why make this change?

This refactoring improves code maintainability and reduces duplication by centralizing query logic. It also adds support for aggregations in both transaction and log queries, enhancing the flexibility of the data retrieval process. The changes make it easier to extend functionality for future query types and improve overall code organization.

### CODE REVIEW

**Comment by AmineAfia** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/48?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#48** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/48?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#44** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/44?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#43** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/indexer/43?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @AmineAfia and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: internal/storage/clickhouse.go [hunk 1]
```diff
@@ -305,170 +307,162 @@ func (c *ClickHouseConnector) GetBlocks(qf QueryFilter) (blocks []common.Block,
...
-		columns, c.cfg.Database, getLimitClause(int(qf.Limit)))
+	return c.executeQuery("transactions", columns, qf, scanTransaction)
+}
+
+func (c *ClickHouseConnector) GetLogs(qf QueryFilter) (QueryResult, error) {
+	columns := "chain_id, block_number, block_hash, block_timestamp, transaction_hash, transaction_index, log_index, address, data, topic_0, topic_1, topic_2, topic_3"
+	return c.executeQuery("logs", columns, qf, scanLog)
+}
+
+func (c *ClickHouseConnector) executeQuery(table, columns string, qf QueryFilter, scanFunc func(driver.Rows) (interface{}, error)) (QueryResult, error) {
```
Comment (iuwqyir) [inline internal/storage/clickhouse.go:320]: You could use a generic type here to have type checking working not only in runtime

**Review by iuwqyir — APPROVED** [review]
LGTM, just one possible improvement with generic types

File: internal/storage/clickhouse.go (hunk 1 above)
Comment (AmineAfia) [reply internal/storage/clickhouse.go]: done

**Review by iuwqyir — APPROVED** [review]

**Comment by AmineAfia** [comment]
### Merge activity

* **Sep 26, 6:12 AM EDT**: @AmineAfia started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thirdweb-dev/indexer/48).
